File: API/src/dtos/files/file.dto.ts
export class FileDto {
  _id?: string = "";
  url: string = "";
  isMain: boolean = false;
  type: string = "";

  public static mapFrom(file: any): FileDto {
    return new FileDto(file);
  }

  private constructor(file: any) {
    this._id = file === null ? "" : file._id;
    this.url = file === null ? "" : file.url;
    this.isMain = file === null ? false : file.isMain;
    this.type = file === null ? "" : file.type;
  }
}

File: frontend/lib/models/file_dto.dart
class FileDto {
  final String? id;
  final String url;
  final bool isMain;
  final String fileType;

  FileDto({
    this.id,
    required this.url,
    required this.isMain,
    required this.fileType,
  });

  factory FileDto.fromJson(Map<String, dynamic> json) {
    return FileDto(
      id: json['id']?.toString(),
      url: json['url']?.toString() ?? '',
      isMain: json['isMain'] ?? false,
      fileType: json['fileType']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'url': url,
      'isMain': isMain,
      'fileType': fileType,
    };
  }
}
